From ba3093cc60e7073486031038e5ac48e08ce6234b Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Sun, 5 Apr 2015 11:40:27 +0200
Subject: Handle wxDateTime data in wxRichTextPrintout::SubstituteKeywords only
 if wxUSE_DATETIME is set to 1.

Code using wxDateTime when preparing printout should be active only if wxUSE_DATETIME == 1.

diff --git a/src/richtext/richtextprint.cpp b/src/richtext/richtextprint.cpp
index 837d3b0..926f1bc 100644
--- a/src/richtext/richtextprint.cpp
+++ b/src/richtext/richtextprint.cpp
@@ -427,10 +427,15 @@ bool wxRichTextPrintout::SubstituteKeywords(wxString& str, const wxString& title
     num.Printf(wxT("%lu"), (unsigned long) pageCount);
     str.Replace(wxT("@PAGESCNT@"), num);
 
+#if wxUSE_DATETIME
     wxDateTime now = wxDateTime::Now();
 
     str.Replace(wxT("@DATE@"), now.FormatDate());
     str.Replace(wxT("@TIME@"), now.FormatTime());
+#else
+    str.Replace(wxT("@DATE@"), wxEmptyString);
+    str.Replace(wxT("@TIME@"), wxEmptyString);
+#endif
 
     str.Replace(wxT("@TITLE@"), title);
 
-- 
1.9.5.msysgit.0

